src/torchmetrics/detection/panoptic_quality.py

```diff
@@ -11,13 +11,12 @@
 # WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
 # See the License for the specific language governing permissions and
 # limitations under the License.
-import warnings
 from typing import Any, Collection, Optional, Sequence, Union
 
 import torch
 from torch import Tensor
 
-from torchmetrics.functional.detection.panoptic_quality import (
+from torchmetrics.functional.detection._panoptic_quality_common import (
```

Comment:
What is the motivation for moving it to a separate file when common still means shared only for this one metric? it generates a much larger diff so it is not so easy to track real changes... :chipmunk:

Reply:
The common functionality is being pulled out into a separate file according to what I mentioned in #1591 . This PR is the first of 2, so that you don’t get hit with an unreviewable one… if there’s a better option let me know!

Reply:
I understand common, but while it is used only by one metric it seems rather specific than common :)

Reply:
Should I call it utils in this PR and rename it to common in the next for consistency? Alternatively all of the common code can stay in panoptic_quality.py and the modified implementation will import from there, but the asymmetry would be slightly confusing

Reply:
it is not about the name, but what other function/metric would use it?

Reply:
The modified PQ metric that is mentioned in the issue and I’m implementing on the side of the standard PQ that’s already there

Reply:
ok, I see now, sorry :innocent:

Reply:
No worries, it was clear we had a misunderstanding but I wasn't sure where 🙂

Reply:
Btw you had a very good reason to be confused, I am now too... https://github.com/Lightning-AI/metrics/issues/1591#issuecomment-1456985706. Let me know if you have a preference either way, we don't need to merge this PR if there's a preference in keeping just 1 metric with 2 flavours.